src/components/chat/chat.js

```diff
@@ -37,91 +30,25 @@ class Chat extends React.Component {
-      mode: "javascript",
-      theme: "default",
-      lineNumbers: true,
-      content: this.textArea.current,
-    })
-    this.codemirror.setSize(null, '80vh');
-    this.codemirror.on('blur', this.codeChanged);
-    this.props.socket.on('editor', (data) => this.codemirror.getDoc().setValue(data.code)); // handles received text
-    this.props.socket.on('newUser', this.updateCode); // this code is not working - what was its purpose?
-
     //HANG-UP
-    this.props.socket.on('hang up', () => {
-      this.openChatModal()
-    })
+    this.props.socket.on('hang up', () => {this.setState({showFeedbackModal: true})})
   }
 
-  //WHY WAS THIS HERE? -- This was added by Arol to prevent the editor from continuously re-rendering
-  // shouldComponentUpdate() {
-  //   return false;
-  // }
-
-  updateCode = (data) => {
-    this.codemirror.getDoc().setValue(data.code);
-    this.setState({keepChangeEditor: this.codemirror.getDoc().getValue()})
-  }
-
-  clickButton = (e) => {
-    e.preventDefault();
-    this.sendMessage();
-  }
-
-  detectEnter = (e) => {
-    var code = e.keyCode || e.which;
-    if (code === 13) {
-      e.preventDefault();
-      this.sendMessage();
-    }
-  }
+  startTimer = () => {
+    setInterval(async () => {
+      this.setState({seconds: this.state.seconds + 1})
+      if (this.state.seconds < 10 ) this.setState({secondsString: '0' + this.state.seconds})
+      else this.setState({secondsString: this.state.seconds})
 
-  sendMessage = () => {
-    const dateNow = new Date();
-    const dateNowFormatted = dateNow.toLocaleString();
-    const msgToSend = {
-      date: dateNowFormatted,
-      id: this.props.socket.id,
-      value: this.message.value,
-      room: this.state.roomId
-    };
-    this.props.socket.emit('chat message', msgToSend);
-    this.message.value = '';
-  }
+      if (this.state.seconds === 60) {
+        await this.setState({ seconds: 0, minutes: this.state.minutes + 1, secondsString: '00'})
+      }
 
-  codeChanged = () => {
-    const editorContent = this.codemirror.getDoc().getValue();
-    const data = { code: editorContent, room: this.state.roomId } // changed property 'text' to 'code' to be more explicit
-    if (editorContent !== this.state.keepChangeEditor) {
-      this.props.socket.emit('editor', data);
-      this.setState({keepChangeEditor: this.codemirror.getDoc().getValue()});
-    }
-    // todo: update the chatmirror so it updates during typing, not just on clicking outside of the box
-    // var myElement = document.getElementById('txtArea');
-    // myElement.focus();
-    // var startPosition = myElement.selectionStart;
-    // var endPosition = myElement.selectionEnd;
-    // console.log('startPosition ', startPosition)
-    // console.log('endPosition ', endPosition)
+      if (this.state.minutes === 15) {
+        this.setState({ overTime: 'red'});
+      }
+  
+    }, 1000)
   }
     
   renderOverlay = () => {
```

Comment:
This renderOverlay should be called renderOverlayModal 

and the overlay.js should be renamed to modal-chat-learner-wait or something more specific, then moved to the modal component folder where we are putting the modals.